mindmeld/auto_annotator.py

```diff
@@ -0,0 +1,691 @@
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[entity["dim"]]
+
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            return
+
+        for candidate in candidates:
+            if (
+                candidate["entity_type"] == entity["dim"]
+                and entity["body"] == candidate["body"]
+            ):
+                entity["value"] = candidate["value"]
+                return entity
+
+    def _resolve_quantity(self, entity):
+        """ Resolves a quantity related entity. First looks for an exact match, then
+        for the largest substring match. Order of priority is "sys_distance" then "sys_quantity".
+        Unresolved entities are labelled as "sys_other-quantity"
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            entity["dim"] = "sys_other-quantity"
+            return entity
+
+        entity_types = ["distance", "quantity"]
+        for entity_type in entity_types:
+            for candidate in candidates:
+                if (
+                    candidate["dim"] == entity_type
+                    and candidate["body"] == entity["body"]
+                ):
+                    entity["value"] = candidate["value"]
+                    entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[
+                        entity_type
+                    ]
+                    return entity
+
+        if SpacyAnnotator._resolve_largest_substring(
+            entity, candidates, entity_types=entity_types, is_time_related=False
+        ):
+            return entity
+        else:
+            entity["dim"] = "sys_other-quantity"
+            return entity
+
+    def _resolve_percent(self, entity):
+        """ Resolves an entity related to percentage. Uses a heuristic of finding
+        the largest candidate value and dividing by 100.
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[entity["dim"]]
+
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            return
+
+        possible_values = []
+        for candidate in candidates:
+            if candidate["entity_type"] == "sys_number":
+                if "." in candidate["body"]:
```

Comment:
I think this function needs to validate that the input `candidate["body"]` is a float. Checking `"." in candidate["body"]` would not cut it since it would pass strings like `alex.bob`. Maybe try this:
```
try:
    float(element)
except ValueError:
    logger.error()
```

Reply:
Since this function filters the candidates to be of type "sys_number", we can expect to receive either an int or float as a string and not a string like "alex.bob". Validating a float with a try-except works well to distinguish between numerical and non-numerical values, however, it doesn't allow us to distinguish between int and float as both would pass the try block.

We can make a slight adjustment and validate an int instead since something like int("3.4") gives a value error